app/controllers/catalog_controller.rb

```diff
@@ -905,15 +909,25 @@ def ansible_playbook_type?
                 elsif @edit
                   @edit[:new][:st_prov_type]
                 end
+    if prov_type == 'generic_ansible_playbook'
+      @current_region = MiqRegion.my_region.region
+    end
     prov_type == 'generic_ansible_playbook'
   end
 
-  def ansible_playbook?
-    ansible_playbook = ansible_playbook_type?
-    @current_region = MiqRegion.my_region.region if ansible_playbook
-    ansible_playbook
+  def terraform_template_type?
+    prov_type = if params[:st_prov_type]
+                  params[:st_prov_type]
+                elsif @record
+                  @record.prov_type
+                elsif @edit
+                  @edit[:new][:st_prov_type]
+                end
+    if prov_type == 'generic_terraform_template'
+      @current_region = MiqRegion.my_region.region
+    end
+    prov_type == 'generic_terraform_template'
```

Comment:
These methods are nearly identical.  Can we try to refactor them?  It's also odd that a predicate/query/boolean method would set the `@current_region` as a side effect.  I understand if this was the existing code for ansible but we should really try to fix that.

Reply:
I can fix this in a follow up pr since I don't want to risk breaking the Ansible Playbook forms and the rest of this pr is pretty much done